packages/quill/src/core/editor.ts

```diff
@@ -370,7 +370,8 @@ function convertHTML(
     return blot.html(index, length);
   }
   if (blot instanceof TextBlot) {
-    return escapeText(blot.value().slice(index, index + length));
+    const escapedText = escapeText(blot.value().slice(index, index + length));
+    return escapedText.replaceAll(' ', '&nbsp;');
```

Comment:
Just curious, why was it necessary? I got an error after update because of this. The text I get from a quill instance using `event.editor.getSemanticHTML()` now returns `&nbsp;` instead of all spaces resulting in wrong behavior when I pass this string to another service, which renders it in PDF.

Reply:
We are seeing the &nbsp; on copy/pasting into the editor. We're using PrimeNG 17.18.12 with quill 2.0.3. We reverted to quill 2.0.2 because of this issue